web/libs/ui/src/lib/select/select.tsx

```diff
@@ -0,0 +1,416 @@
+import React, { type ForwardedRef, forwardRef, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from "@humansignal/shad/components/ui/command";
+import { Popover, PopoverContent, PopoverTrigger } from "@humansignal/shad/components/ui/popover";
+import type { SelectOption, OptionProps, SelectProps } from "./types.ts";
+import { Checkbox, Label } from "@humansignal/ui";
+import { isDefined } from "@humansignal/core/lib/utils/helpers";
+import { IconChevron, IconChevronDown } from "@humansignal/icons";
+import clsx from "clsx";
+import styles from "./select.module.scss";
+
+export const Select = forwardRef(
+  <T, A extends SelectOption<T>[]>(
+    {
+      label,
+      description,
+      options = [],
+      validate,
+      required,
+      skip,
+      labelProps,
+      defaultValue,
+      searchable,
+      searchPlaceholder,
+      value: externalValue,
+      disabled = false,
+      multiple = false,
+      isInline = false,
+      isInProgress = false,
+      triggerProps,
+      className,
+      size,
+      searchFilter,
+      onSearch,
+      selectedValueRenderer,
+      ...props
+    }: SelectProps<T, A>,
+    _ref: ForwardedRef<HTMLSelectElement>,
+  ) => {
+    const ref = _ref ?? useRef<HTMLSelectElement>();
+    const triggerRef = useRef<HTMLDivElement>();
+    const [query, setQuery] = useState<string>("");
+    const valueRef = useRef<any>();
+    let initialValue = defaultValue?.value ?? defaultValue ?? externalValue?.value ?? externalValue;
+
+    if (multiple) {
+      initialValue = initialValue ? (Array.isArray(initialValue) ? (initialValue ?? []) : [initialValue]) : [];
+    } else if (Array.isArray(initialValue)) {
+      initialValue = initialValue[0];
+    }
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const [value, setValue] = useState<any>(initialValue);
+
+    valueRef.current = value;
+    useEffect(() => {
+      if (!isDefined(externalValue)) return;
+      let val = externalValue?.value ?? externalValue;
+      if (multiple && !Array.isArray(val)) {
+        val = [val];
+      } else if (!multiple && Array.isArray(val)) {
+        val = val[0];
+      }
+      setValue(val);
+    }, [externalValue, multiple]);
+
+    useEffect(() => {
+      if (!isOpen) setQuery("");
+    }, [isOpen]);
+    const _onChange = useCallback(
+      (val: string, isSelected: boolean) => {
+        if (disabled) return;
+
+        let selectedOptions = [];
+        if (multiple) {
+          setValue((prev = []) => {
+            if (isSelected) {
+              return (selectedOptions = [...prev.filter((v) => v !== val)]);
+            }
+            return (selectedOptions = [...prev, val]);
+          });
```

Comment:
this could cause an issue with the callback handler getting called after passing selectedOptions up to props.onChange. use a valueRef instead of the callback